src/back/msl.rs

```diff
@@ -819,8 +813,10 @@ impl<W: Write> Writer<W> {
 
                 if em != spirv::ExecutionModel::GLCompute {
                     writeln!(self.out, "struct {} {{", location_input_name)?;
-                    for &handle in var_inputs.iter() {
-                        let var = &module.global_variables[handle];
+                    for ((handle, var), &usage) in module.global_variables.iter().zip(&fun.global_usage) {
```

Comment:
`module.global_variables.iter().zip(&fun.global_usage)` is used four times in this function. Isn't it possible to do this once and put it into a variable, to reduce iterations through `global_variables` and prevent to zip four times?

Reply:
we could do it, yeah, although it's not going to save us any lines :)

Reply:
Less lines are not always better :) I was pointing it out more as a performance optimisation. I am not sure if there is a need for it though.

Reply:
ah ok. There isn't any performance impact here. It's just an iteration over a few dozen of globals, super light.

Reply:
Sure, then it is all fine by me, rest looks good 👍